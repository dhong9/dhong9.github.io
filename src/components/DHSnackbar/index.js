import { forwardRef } from "react";

import Snackbar from "@mui/material/Snackbar";
import MuiAlert from "@mui/material/Alert";
import Slide from "@mui/material/Slide";

// prop-types is a library for typechecking of props
import PropTypes from "prop-types";

const TransitionLeft = (props) => <Slide {...props} direction="left" />;

const Alert = forwardRef((props, ref) => (
  <MuiAlert elevation={6} ref={ref} variant="filled" {...props} />
));

export default function DHSnackbar({ onClose, open, severity, message }) {
  return (
    <Snackbar
      TransitionComponent={TransitionLeft}
      anchorOrigin={{ vertical: "top", horizontal: "right" }}
      open={open}
      autoHideDuration={6000}
      onClose={onClose}
    >
      <Alert onClose={message} severity={severity} sx={{ width: "100%" }}>
        {message}
      </Alert>
    </Snackbar>
  );
}

// Typechecking props of DHSnackbar
DHSnackbar.propTypes = {
  open: PropTypes.bool,
  onClose: PropTypes.func,
  severity: PropTypes.string,
  message: PropTypes.string,
};

DHSnackbar.defaultProps = {
  open: false,
  onClose: null,
  severity: "",
  message: "",
};
