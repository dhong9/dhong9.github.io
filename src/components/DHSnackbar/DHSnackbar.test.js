import renderer from "react-test-renderer";
import DHSnackbar from "components/DHSnackbar";

// Mocks
jest.mock("@mui/material/Snackbar", () => {
  const { forwardRef } = jest.requireActual("react");
  const Snackbar = forwardRef((props, ref) => (
    <div ref={ref}>
      Mock Snackbar
      {props.children}
    </div>
  ));
  return {
    __esModule: true,
    default: Snackbar,
  };
});
jest.mock("@mui/material/Alert", () => {
  const { forwardRef } = jest.requireActual("react");
  const Alert = forwardRef((props, ref) => (
    <div ref={ref}>
      Mock Alert
      {props.children}
    </div>
  ));
  return {
    __esModule: true,
    default: Alert,
  };
});
jest.mock("@mui/material/Slide", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <div>Mock Slide</div>),
  };
});

describe("DHSnackbar", () => {
  it("renders", () => {
    const component = renderer.create(<DHSnackbar />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
