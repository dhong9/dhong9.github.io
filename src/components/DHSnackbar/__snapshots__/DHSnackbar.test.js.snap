// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DHSnackbar renders 1`] = `
<div>
  Mock Snackbar
  <div>
    Mock Alert
  </div>
</div>
`;
